pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -264,36 +261,133 @@ func (sync Synchronizer) syncOperationalState(errChan chan<- events.DeviceRespon
 	// TODO Parse the GET result to get actual instances of list items (replacing
 	//  wildcarded paths - assuming that stratum does not support wildcarded GET
 	//  and SUBSCRIBE)
+	log.Infof("%d ReadOnly paths for %s", len(sync.modelReadOnlyPaths), sync.key.DeviceID)
+	if len(sync.modelReadOnlyPaths) == 0 {
+		noPathErr := fmt.Errorf("target %#v has no paths to subscribe to", sync.ID)
+		errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorSubscribe,
+			sync.key.DeviceID, noPathErr)
+		log.Warning(noPathErr)
+		return
+	}
+
+	wildExpandedPaths := make([]string, 0)
+	if len(notifications) == 0 {
+		log.Infof("No notifications received - trying Get on the readonly paths instead %s", sync.key.DeviceID)
+		getPaths := make([]*gnmi.Path, 0)
+		for _, path := range modelregistry.Paths(sync.modelReadOnlyPaths) {
+			gnmiPath, err := utils.ParseGNMIElements(utils.SplitPath(path))
+			if err != nil {
+				log.Warning("Error converting RO path to gNMI")
+				errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorTranslation,
+					sync.key.DeviceID, err)
+				return
+			}
+			getPaths = append(getPaths, gnmiPath)
+		}
+
+		requestRoPaths := &gnmi.GetRequest{
+			Encoding: sync.encoding,
+			Path:     getPaths,
+		}
+
+		responseRoPaths, errRoPaths := target.Get(target.Ctx, requestRoPaths)
+		if errRoPaths != nil {
+			log.Warning("Error on request for read-only paths", sync.key, errRoPaths)
+			errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorGetWithRoPaths,
+				sync.key.DeviceID, errRoPaths)
+			return
+		}
+
+		notifications = append(notifications, responseRoPaths.Notification...)
+
+		////////////////////////////////////////////////////////////////////////
+		// Special case for Stratum Aug'19 - the notification will only contain
+		// the ifindex and name of the interface - to get the other state attributes
+		// it has to be called again with the wildcard values expanded out. This
+		// is because Stratum does not support wildcards properly.
+		// Fortunately these can be also used for subscribe
+		////////////////////////////////////////////////////////////////////////
+		if sync.Type == "Stratum" {
```

Comment:
I'm wondering if we ought to implement some form of driver mechanism. Since this type of device define behaviour is allowed in gNMI since it follows the gNMI spec, other devices can follow different behaviour and this class woudl have to account for all of them @tomikazi thoughts on this ?